src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
```

Comment:
how about renaming this method to `max()`? this would be readable for me:

```
int max = this.max(talks); // "max is the maximum of all talks"
```

Don't you think? In general, avoid long composite names, they only make noise. The code should be readable by itself.